python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -148,7 +151,25 @@ def _sample_subgraphs(self, seeds):
         -------
         Union[torch.Tensor, Dict[str, torch.Tensor]]
             The input nodes.
-        SampledSubgraph
+        List[SampledSubgraph]
             The sampled subgraphs.
+
+        Examples
+        --------
+        >>> @functional_datapipe("my_sample_subgraph")
+        >>> class MySubgraphSampler(SubgraphSampler):
+        >>>     def __init__(self, datapipe, graph, fanouts):
+        >>>         super().__init__(datapipe)
+        >>>         self.graph = graph
+        >>>         self.fanouts = fanouts
+        >>>     def sample_subgraphs(self, seeds):
+        >>>         # Sample subgraphs from the given seeds.
+        >>>         subgs = []
+        >>>         for fanout in reversed(self.fanouts):
+        >>>             # Sample a fixed number of neighbors for each seed node.
+        >>>             input_nodes, subg = self.graph.sample_neighbors(seeds, fanout)
+        >>>             seeds = input_nodes
+        >>>             subgs.insert(0, subg)
+        >>>         return input_nodes, subgs
```

Comment:
```suggestion
        >>>         # Sample subgraphs from the given seeds.
        >>>         subgraphs = []
        >>>         subgraphs_nodes = {}
        >>>         for fanout in reversed(self.fanouts):
        >>>             subgraph = self.graph.sample_neighbors(seeds, fanout)
        >>>             subgraphs.insert(0, subgraph)
        >>>             subgraphs_nodes.update(subgraph.nodes)
        >>>             seeds = subgraph.nodes
        >>>         return subgraphs.nodes, subgraphs
```

Reply:
As `input_nodes` is confusing, then we should only return subgraphs only for generality.

Reply:
But we need subgraphs_nodes for feature fetching in later stage.

Reply:
The API here could be further polished.

Reply:
all right.